### PR TITLE
Fixes vulnerabilities on 3.12.z

### PR DESCRIPTION
**Pull request by mmedenjak**
Backport of:
https://github.com/hazelcast/hazelcast/pull/17484
https://github.com/hazelcast/hazelcast/pull/17446

Fixes: 
https://github.com/hazelcast/hazelcast/issues/17444
https://github.com/hazelcast/hazelcast/issues/17425

EE: https://github.com/hazelcast/hazelcast-enterprise/pull/3765

### CODE REVIEW

**Comment by hz-highfive** [comment]
The job `Hazelcast-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/2274/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/2274/Hazelcast-pr-builder-2274.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
---------SUMMARY----------
--------------------------
[ERROR] Failed to execute goal org.codehaus.mojo:animal-sniffer-maven-plugin:1.15:check (source-java6-check) on project hazelcast: Execution source-java6-check of goal org.codehaus.mojo:animal-sniffer-maven-plugin:1.15:check failed. IllegalArgumentException -> [Help 1]
--------------------------
</details>
</pre>

**Comment by hz-highfive** [comment]
The job `Hazelcast-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/2275/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/2275/Hazelcast-pr-builder-2275.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
---------SUMMARY----------
--------------------------
[ERROR] Failed to execute goal org.codehaus.mojo:animal-sniffer-maven-plugin:1.15:check (source-java6-check) on project hazelcast: Execution source-java6-check of goal org.codehaus.mojo:animal-sniffer-maven-plugin:1.15:check failed. IllegalArgumentException -> [Help 1]
--------------------------
</details>
</pre>

**Comment by hz-highfive** [comment]
The job `Hazelcast-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/2276/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/2276/Hazelcast-pr-builder-2276.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
---------SUMMARY----------
--------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:2.15:checkstyle (default) on project hazelcast: An error has occurred in Checkstyle report generation. Failed during checkstyle execution: There is 1 error reported by Checkstyle 6.1.1 with /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/checkstyle/checkstyle.xml ruleset. -> [Help 1]
--------------------------
</details>
</pre>

**Comment by hz-highfive** [comment]
The job `Hazelcast-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/2277/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/2277/Hazelcast-pr-builder-2277.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
---------SUMMARY----------
--------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:2.15:checkstyle (default) on project hazelcast-build-utils: An error has occurred in Checkstyle report generation. Failed during checkstyle execution: There is 1 error reported by Checkstyle 6.1.1 with /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/checkstyle/checkstyle.xml ruleset. -> [Help 1]
--------------------------
</details>
</pre>

File: pom.xml [hunk 1]
```diff
@@ -62,9 +62,9 @@
         <maven.assembly.plugin.version>2.4</maven.assembly.plugin.version>
         <maven.rar.plugin.version>2.2</maven.rar.plugin.version>
         <maven.bundle.plugin.version>2.4.0</maven.bundle.plugin.version>
-        <maven.shade.plugin.version>3.0.0</maven.shade.plugin.version>
+        <maven.shade.plugin.version>3.2.4</maven.shade.plugin.version>
         <maven.dependency.plugin.version>2.10</maven.dependency.plugin.version>
-        <maven.animal.sniffer.plugin.version>1.15</maven.animal.sniffer.plugin.version>
+        <maven.animal.sniffer.plugin.version>1.19</maven.animal.sniffer.plugin.version>
```
Comment (mmedenjak) [inline pom.xml:67]: Without upgrading the animal-sniffer plugin, it failed with a cryptic `InvalidArgumentException` with no reference to what exactly failed. Afterwards, the shade plugin failed so I upgraded it, which caused the change in the method signature for `HazelcastManifestTransformer`

File: pom.xml [hunk 2]
```diff
@@ -82,7 +82,7 @@
         <log4j2.version>2.3</log4j2.version>
         <slf4j.api.version>1.7.25</slf4j.api.version>
 
-        <jackson.version>2.9.7</jackson.version>
+        <jackson.version>2.11.2</jackson.version>
```
Comment (mmedenjak) [inline pom.xml:85]: The two actual dependency upgrades.

**Comment by kwart** [comment]
Do we have also the EE counterpart for hazelcast/hazelcast-enterprise#3762?

**Comment by mmedenjak** [comment]
@kwart yes, there is now (https://github.com/hazelcast/hazelcast-enterprise/pull/3765), haven't gotten around to opening it yesterday. I guess I'll also have to bump up the maven shade and animal sniffer dependencies on 4.x now, since it's strange that the 3.12.z branch has newer dependencies than 4.x.

**Review by kwart — APPROVED** [review]

**Review by blazember — APPROVED** [review]

**Comment by mmedenjak** [comment]
Thank you for the reviews, guys!

**Pull request merged**